src/IECoreImage/DisplayDriverServer.cpp

```diff
@@ -423,9 +437,39 @@ void DisplayDriverServer::Session::handleReadOpenParameters( const boost::system
 
 		const StringData *displayType = parameters->member<StringData>( "remoteDisplayType", true /* throw if missing */ );
 
+		const BoolData *mergeDriverData = parameters->member<BoolData>( "mergeDriver", false);
+        m_mergeSession = mergeDriverData && mergeDriverData->readable();
+
 		// create a displayDriver using the factory function.
-		m_displayDriver = DisplayDriver::create( displayType->readable(), displayWindow->readable(), dataWindow->readable(), channelNames->readable(), parameters );
+		if (!m_mergeSession)
+		{
+			m_displayDriver = DisplayDriver::create( displayType->readable(), displayWindow->readable(), dataWindow->readable(), channelNames->readable(), parameters );
+		}
+		else
+		{
+			m_mergeId = parameters->member<IntData>( "sessionId", true /* throw if missing */ )->readable();
 
+			// Check if merge ID in map, if not then create display driver and session count pair with merge ID.
+			if (const auto search = mergeMap.find(m_mergeId); search == mergeMap.end())
```

Comment:
As above, I'm not a fan of packing a lot of stuff into an `if()`, although in this case I suppose it does limit the scope of the `search` variable, which is nice.

But as above, we're doing more map searches than necessary - `find()`does a search, then `emplace()` does a search and then `[]` does a search. Since in all cases we want to end up with an entry in the map, we could just do this :

```
auto &m = mergeMap[m_mergeId];
if( !m.first )
{
    // First driver
    m.first = DisplayDriver::create(
    ...
}
```

Reply:
I don't know enough about DisplayDriverServer to know if different sessions get handled on different threads. If they do, then we need to consider thread-safety for all accesses to the `mergeMap`. Do you know if this is an issue or not, or is everything on a single thread?

Reply:
From my understanding the server runs as in a separate thread to the rest of cortex, but handles all accepts in a single threaded manner. In which case we wouldn't have to worry about thread-safety.